include/stdexec/__detail/__basic_sender.hpp

```diff
@@ -73,10 +77,239 @@ namespace stdexec {
     STDEXEC_PRAGMA_POP()
 
     struct __tie {
-      template <class _Tag, class _Data, class... _Children>
-      constexpr auto operator()(_Tag, _Data&& __data, _Children&&... __children) const noexcept {
-        return std::tuple<_Tag, _Data&&, _Children&&...>{
-          {}, (_Data&&) __data, (_Children&&) __children...};
+      template <class _Tag, class _Data, class... _Child>
+      STDEXEC_ATTRIBUTE((always_inline))
+      constexpr auto operator()(_Tag, _Data&& __data, _Child&&... __child) const noexcept {
+        return std::tuple<_Tag, _Data&&, _Child&&...>{{}, (_Data&&) __data, (_Child&&) __child...};
+      }
+    };
+
+    // Note: This is UB. UBSAN allows it for now.
+    template <class _Parent, class _Child>
+    _Parent* __parent_from_child(_Child* __child, _Child _Parent::*__mbr_ptr) noexcept {
+      alignas(_Parent) char __buf[sizeof(_Parent)];
+      _Parent* __parent = (_Parent*) &__buf;
+      const std::ptrdiff_t __offset = (char*) &(__parent->*__mbr_ptr) - __buf;
+      return (_Parent*) ((char*) __child - __offset);
+    }
+
+    template <class _Sexpr, class _Receiver>
+    struct __op_state;
+
+    template <class _Receiver, class _Sexpr, class _Idx>
+    struct __receiver {
+      using is_receiver = void;
+      using __sexpr = _Sexpr;
+      using __index = _Idx;
+
+      // A pointer to the parent operation state, which contains the one created with
+      // this receiver.
+      __op_state<_Sexpr, _Receiver>* __op_;
+
+      template <class _ChildSexpr, class _ChildReceiver>
+      static __receiver __from_op_state(__op_state<_ChildSexpr, _ChildReceiver>* __child) noexcept {
+        using __parent_op_t = __op_state<_Sexpr, _Receiver>;
+        std::ptrdiff_t __offset = __parent_op_t::template __get_child_op_offset<__v<_Idx>>();
+        __parent_op_t* __parent = (__parent_op_t*) ((char*) __child - __offset);
+        return __receiver{__parent};
+      }
+
+      template <__completion_tag _Tag2, class... _Args>
+      STDEXEC_ATTRIBUTE((always_inline))
+      friend void tag_invoke(_Tag2, __receiver&& __self, _Args&&... __args) noexcept {
+        __self.__op_->__complete(_Tag2(), (_Args&&) __args...);
+      }
+
+      template <same_as<get_env_t> _Tag2>
+      STDEXEC_ATTRIBUTE((always_inline))
+      friend auto tag_invoke(_Tag2, const __receiver& __self) noexcept -> env_of_t<_Receiver> {
```

Comment:
This is another "gift" of `tag_invoke`. All overloads are considered, even ones with mismatching tags, because maybe the compiler finds a conversion from `connect_t` to `get_env_t`, say. So the compiler charges ahead computing the return type and then hard fails for one reason or another.

We avoid that by deducing the tag type and aborting early if it isn't the one we're looking for. Ridiculous, I know.